src/meta/src/stream/scale/reschedule.rs

```diff
@@ -150,7 +141,7 @@ pub(crate) fn rebalance_actor_vnode(
         target_actor_count,
     );
 
-    let (mut removed, mut rest): (Vec<_>, Vec<_>) = actors
+    let (removed, rest): (Vec<_>, Vec<_>) = actors
```

Comment:
I don't see `removed` vnodes being used if `actors_to_remove.len() != actors_to_create.len()`. How do we reassign the removed vnodes? I know the logic might be in `divide_vnode_into_occupied_and_vacant` and `appoint_actors_to_vacant_interval_sets` but I suggest we give an concrete example on how the algorithm works to help better understand it in high level. Also, we can also list out the sceanrios when the cache invalidation (caused by more data movement) is larger than the previous algorithm to help understand the tradeoff between cache invalidation and cross-vnode locality in the current algorithm.